docs/architecture/api-packages.md

```diff
@@ -0,0 +1,156 @@
+# API Package
+
+## Introduction
+
+Previously, in Kyma Runtime, single API or Event Definition of a given Application was represented as a single Service Class in Service Catalog.
+
+To have control over this behavior, we introduce API Packages. API Package is a way of grouping multiple API or Event Definitions, so that they are represented as single Service Class.
+
+## Solution
+
+![API Packages Diagram](./assets/api-packages.svg)
+
+In order to create API packages, Compass utilize Application labels. On Application level, one can specify package by using label in a form of the following key and value:
+
+**Key:** `api-packages`
+**Value:**
+
+```json
+{
+  "foo": {
+    "description": "",
+    "metadata": {
+      "displayName": "",
+      "documentationURL": "",
+      "supportURL": ""
+    },
+    "events": [
+      "707ea9da-0bf8-4dad-8145-a934ee1ebffb",
+      "6e5969c6-63fd-42c0-af17-72ed95c9dcd5"
+    ],
+    "apis": ["foo", "bar"]
+  }
+}
+```
+
+The root object key defines API Package name. In this example, `foo` is the API Package name.
+
+In order to allow validation of the label value, Compass creates `api-packages` LabelDefinition automatically. It is done while creating new Application. Once Tenant management feature is ready, creating new tenant will be used as a trigger for the LabelDefinition creation.
+The `api-packages` LabelDefinition has the following JSON schema:
+
+```json
```

Comment:
Something that is worth considering here is the fact that we allow the user to edit this JSON schema. I think we need to introduce some concepts that will allow us defining LabelDefinitions that will not be editable by the users.

Reply:
Right, I forgot to mention that. I will update this paragraph.